src/pkg/podman/podman.go

```diff
@@ -0,0 +1,312 @@
+package podman
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/HarryMichal/go-version"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+var (
+	// ErrNonExistent signals one of the specified containers does not exist (applies to `podman rm/rmi`)
+	ErrNonExistent = errors.New("exit status 1")
+	// ErrRunningContainer signals one of the specified containers is paused or running (applies to `podman rm`)
+	ErrRunningContainer = errors.New("exit status 2")
+	// ErrHasChildren signals one of the specified images has child images or is used by a container (applies to `podman rmi`)
+	ErrHasChildren = errors.New("exit status 2")
+	// ErrInternal signals an error in Podman itself
+	ErrInternal = errors.New("exit status 125")
+	// ErrCmdCantInvoke signals a contained command cannot be invoked (applies to `podman run/exec`)
+	ErrCmdCantInvoke = errors.New("exit status 126")
+	// ErrCmdNotFound signals a contained command cannot be found (applies to `podman run/exec`)
+	ErrCmdNotFound = errors.New("exit status 127")
+	// ErrServiceUnavailable signals a problem while pulling an image from a registry (applies to `podman pull`)
+	ErrServiceUnavailable = errors.New("invalid status code from registry 503 (Service Unavailable)")
+	// ErrConnectionRefused signals the host machine is probably not online (applies to `podman pull`)
+	ErrConnectionRefused = errors.New("connection refused")
+	// ErrUnknownManifest signals the requested image does not exist (applies to `podman pull`)
+	ErrUnknownManifest = errors.New("manifest unknown")
+)
+
+func IsPathBindMount(path string, containerInfo map[string]interface{}) bool {
+	containerMounts := containerInfo["Mounts"].([]interface{})
+	for _, mount := range containerMounts {
+		dest := fmt.Sprint(mount.(map[string]interface{})["Destination"])
+		if dest == path {
+			return true
+		}
+	}
+
+	return false
+}
+
+// CheckVersion compares provided version with the version of Podman.
+//
+// Takes in one string parameter that should be in the format that is used for versioning (eg. 1.0.0, 2.5.1-dev).
+//
+// Returns 0 if required version equals Podman version
+// Returns 1 if required version is higher than Podman version
+// Returns -1 if required version is lower than Podman version
+func CheckVersion(requiredVersion string) int {
```

Comment:
While it's nice that the underlying `version` package follows the same convention as C's `strcmp`, I think `CheckVersion` should be a simpler boolean check like [GLIB_CHECK_VERSION](https://developer.gnome.org/glib/stable/glib-Version-Information.html#GLIB-CHECK-VERSION:CAPS).

The vast majority of times (possibly always?) we are only interested in checking if Podman is new enough or not, and we don't really care whether the versions exactly match or not. This means we no longer care about a tristate reply, but just a boolean is enough.

If nothing else, the code becomes easier to read. :)